src/widgets/myspeed/component.jsx

```diff
@@ -0,0 +1,64 @@
+import { useTranslation } from "next-i18next";
+
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+export default function Component({ service }) {
+  const { t } = useTranslation();
+  const { widget } = service;
+  const { data, error } = useWidgetAPI(widget, "info");
+
+  if (error) {
+    return <Container service={service} error={error} />;
+  }
```

Comment:
```suggestion
  if (error || (data && data.message) || (data && data[0] && data[0].error)) {
    let finalError = error ?? data;
    if (data && data[0] && data[0].error) {
      try {
        finalError = JSON.parse(data[0].error);
      } catch (e) {
        finalError = data[0].error;
      }
    }
    return <Container service={service} error={finalError} />;
  }
```

I think cleaner to put all the error handling together. At least in my basic testing this covered it:
<img width="752" alt="Screenshot 2024-06-23 at 9 59 41 AM" src="https://github.com/gethomepage/homepage/assets/4887959/23dc1919-d5e3-4ec5-aa42-46158d101b57">